dashboard/src/components/Header/ContextSelector.tsx

```diff
@@ -57,7 +59,14 @@ function ContextSelector() {
     setStateNamespace(event.target.value);
   const changeContext = () => {
     dispatch(actions.namespace.setNamespace(cluster, namespace));
-    dispatch(push(app.apps.list(cluster, namespace)));
+    // Regex matching a namespaced route: e.g. /c/cluster/ns/namespace
+    const nsRegex = /^\/c\/([^/]*)\/ns\/[^/]*\//;
+    if (nsRegex.test(location.pathname)) {
+      // Change the namespace in the route
+      history.push(
+        location.pathname.replace(nsRegex, `/c/$1/ns/${namespace}/`).concat(location.search),
```

Comment:
Nice. I was wondering how the two substitutions are treated differently, but see the braces are required for template literals but it'll ignore the `$1` without braces.